apps/dapp-console/app/settings/components/IneligibleRebateDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import { User } from '@privy-io/react-auth'
+import { useCallback, useMemo, useState } from 'react'
+import { Transaction } from 'viem'
+
+import { Text } from '@eth-optimism/ui-components/src/components/ui/text/text'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTrigger,
+} from '@eth-optimism/ui-components'
+
+export type IneligibleRebateReason =
+  | 'deployed-before-account-creation'
+  | 'max-rebate-amount-claimed'
+
+export type IneligibleRebateDialogProps = {
+  reason: IneligibleRebateReason
+  user: User | null
+  contractDeploymentTx?: Transaction & {
```

Comment:
placeholder type, this will turn into the transaction obj we'll have in the backend